core/arch/arm/kernel/thread_spmc.c

```diff
@@ -956,9 +1041,15 @@ static int handle_mem_share_rxbuf(size_t blen, size_t flen,
 		goto out;
 	}
 
+	if (IS_ENABLED(CFG_NS_VIRTUALIZATION) &&
+	    virt_set_guest(mem_trans.sender_id))
```

Comment:
Shouldn't we add this in `handle_mem_share_tmem()` too? Before calling `add_mem_share()` from that function, just like here.

Reply:
You're right, I'll fix it.